Sources/BSON/BSONObjectID.swift

```diff
@@ -69,6 +69,28 @@ extension BSONObjectID: BSONValue {
     internal func write(to buffer: inout ByteBuffer) {
         buffer.writeBytes(self.oid)
     }
+
+    public init(from decoder: Decoder) throws {
+        // assumes that the BSONObjectID is stored as a valid hex string.
+        let container = try decoder.singleValueContainer()
+        let hex = try container.decode(String.self)
+        guard let oid = try? BSONObjectID(hex) else {
+            throw DecodingError.dataCorrupted(
+                DecodingError.Context(
+                    codingPath: decoder.codingPath,
+                    debugDescription: "Invalid ObjectID hex string. Got: \(hex)"
```

Comment:
might want to just use the error message string here now that we throw more descriptive messages

Reply:
done